components/post_view/post_add_bot_teams_channels/post_add_bot_teams_channels.jsx

```diff
@@ -0,0 +1,197 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import PropTypes from 'prop-types';
+import React from 'react';
+import {Button} from 'react-bootstrap';
+import {FormattedMessage} from 'react-intl';
+
+import FormattedMarkdownMessage from 'components/formatted_markdown_message';
+import GenericTeamChannelProvider from 'components/suggestion/generic_team_channel_provider.jsx';
+import MenuActionProvider from 'components/suggestion/menu_action_provider';
+import AutocompleteSelector from 'components/widgets/settings/autocomplete_selector';
+import PostEmoji from 'components/post_emoji';
+
+import {localizeMessage} from 'utils/utils.jsx';
+
+export default class PostAddBotTeamsChannels extends React.PureComponent {
+    static propTypes = {
+
+        /*
+        * Post containing the user id of the bot
+        */
+        post: PropTypes.object.isRequired,
+
+        /*
+        * List of teams available to the current user
+        */
+        teams: PropTypes.arrayOf(PropTypes.object).isRequired,
+
+        actions: PropTypes.shape({
+            addChannelMember: PropTypes.func.isRequired,
+            addUserToTeam: PropTypes.func.isRequired,
+            editPost: PropTypes.func.isRequired,
+        }).isRequired,
+    }
+
+    constructor(props) {
+        super(props);
+        this.state = {addMode: false};
+    }
+
+    handleAddButton = () => {
+        this.setState({addMode: true});
+    }
+
+    handleSelectedTeam = (selected) => {
+        this.setState({selectedTeamName: selected.text, team: selected});
+    }
+
+    handleSelectedChannel = async (selected) => {
+        this.setState({selectedChannelName: selected.display_name, channel: selected});
+
+        await this.props.actions.addUserToTeam(this.state.team.value, this.props.post.user_id);
+        await this.props.actions.addChannelMember(this.state.channel.id, this.props.post.user_id);
+
+        // Check to see if we've already added the bot to the channel, it doesn't need to show up twice
+        var currentAddedChannels = this.props.post.props.addedChannels;
+        if (currentAddedChannels) {
+            var channelAlreadyExists = currentAddedChannels.find((c) => c.id === this.state.channel.id);
+            if (!channelAlreadyExists) {
+                currentAddedChannels = currentAddedChannels.concat([{
+                    id: this.state.channel.id,
+                    name: this.state.selectedChannelName,
+                }]);
+            }
+        } else {
+            currentAddedChannels = [{
+                id: this.state.channel.id,
+                name: this.state.selectedChannelName,
+            }];
+        }
+
+        const post = {
+            id: this.props.post.id,
+            props: {
+                addedChannels: currentAddedChannels,
+            },
+        };
+
+        await this.props.actions.editPost(post);
```

Comment:
Since we're not actually doing anything after the `await` here you can just call `this.props.actions.editPost(post);` without the `await`

Reply:
Done!